### PR TITLE
Added findAndReplaceInModels

### PR DESCRIPTION
**Pull request by itsmepvr**
Hey @FooSoft ,
### Added `findAndReplaceInModels` api call.
* To find and replace string in existing models by name or all models by name - 'all_models'.
* Also added needed documentation in models.
* Referred and Implemented this [addon](https://ankiweb.net/shared/info/992153575).

### CODE REVIEW

File: plugin/__init__.py [hunk 1]
```diff
@@ -918,6 +918,41 @@ def updateModelStyling(self, model):
         models.flush()
 
 
+    @util.api()
+    def findAndReplaceInModels(self, modelName, findText, replaceText, Front = True, Back = True, css = True):
+        if modelName == 'all_models':
```
Comment (FooSoft) [inline plugin/__init__.py:923]: Just make `modelName` be optional; if not provided, it means all models.

File: plugin/__init__.py [hunk 2]
```diff
@@ -918,6 +918,41 @@ def updateModelStyling(self, model):
...
+                        if Back and key == 'afmt' and findText in value:
+                            checkForText = True
+                            tmpls[key] = value.replace(findText, replaceText)                       
+            self.collection().models.save(model, True)
+            self.collection().models.flush()
+            if checkForText:
+                updatedModels += 1
+        if updatedModels == len(ankiModel):
+            return "Found and replaced in {0} models".format(updatedModels)
+        else:
```
Comment (FooSoft) [inline plugin/__init__.py:952]: Don't need `else`

File: plugin/__init__.py [hunk 3]
```diff
@@ -918,6 +918,41 @@ def updateModelStyling(self, model):
         models.flush()
 
 
+    @util.api()
+    def findAndReplaceInModels(self, modelName, findText, replaceText, Front = True, Back = True, css = True):
```
Comment (FooSoft) [inline plugin/__init__.py:922]: Don't do uppercase for `Front` and `Back`, no spaces around `=` in argument lists.

File: plugin/__init__.py [hunk 4]
```diff
@@ -918,6 +918,41 @@ def updateModelStyling(self, model):
...
+        for model in ankiModel:
+            model = self.collection().models.byName(modelName)
+            checkForText = False
+            if css and findText in model['css']:
+                checkForText = True
+                model['css'] = model['css'].replace(findText, replaceText) 
+            for tmpls in model.get('tmpls'):
+                for key, value in tmpls.items():
+                    if key in ('qfmt', 'afmt'):
+                        if Front and key == 'qfmt' and findText in value:
```
Comment (FooSoft) [inline plugin/__init__.py:940]: Do explicit checks `is None` or `is not None` for `front` and `back`.

File: plugin/__init__.py [hunk 5]
```diff
@@ -918,6 +918,41 @@ def updateModelStyling(self, model):
...
+                            tmpls[key] = value.replace(findText, replaceText)
+                        if Back and key == 'afmt' and findText in value:
+                            checkForText = True
+                            tmpls[key] = value.replace(findText, replaceText)                       
+            self.collection().models.save(model, True)
+            self.collection().models.flush()
+            if checkForText:
+                updatedModels += 1
+        if updatedModels == len(ankiModel):
+            return "Found and replaced in {0} models".format(updatedModels)
```
Comment (FooSoft) [inline plugin/__init__.py:951]: Why return a formatted string? That can be annoying for caller that may want to use the value. Just return the count.

**Review by FooSoft — CHANGES_REQUESTED** [review]
Also please add unit test.

File: tests/test_models.py [hunk 6]
```diff
@@ -57,4 +57,4 @@ def runTest(self):
 
 
 if __name__ == '__main__':
-    unittest.main()
+    unittest.main()
```
Comment (FooSoft) [inline tests/test_models.py:60]: Was there meant to be a test added in this file?

Comment (itsmepvr) [reply tests/test_models.py:60]: Yes, still need to do it. Actually I never wrote unit tests before. Will do it.

Comment (itsmepvr) [reply tests/test_models.py:66]: @FooSoftm Hey, I added unit test, let me know if that is correct cause I am a newbie writing unit tests.

Comment (itsmepvr) [reply tests/test_models.py:66]: Added unit tests. Let me know if it is correct cause I am a newbie for writing unit tests. Thanks

File: plugin/__init__.py (hunk 1 above)
Comment (FooSoft) [reply plugin/__init__.py:923]: No I mean just make it `None`, this way if someone has a `modelName` of `all_models` it won't be an issue. There's no reason to make this be an actual special string. Also, I see that `findText` and `replaceText` default to empty string, why? I think if you are calling `findAndReplaceInModels` it's reasonable to assume that `findText` and `replaceText` are provided.

File: tests/test_models.py (hunk 6 above)
Comment (FooSoft) [reply tests/test_models.py:66]: Tests themselves look fine I think. 👍🏻

**Review by FooSoft — CHANGES_REQUESTED** [review]

File: plugin/__init__.py (hunk 1 above)
Comment (itsmepvr) [reply plugin/__init__.py]: Updated..!! Please check

**Pull request merged**